File: code/generate_strategies/IR(pseudocode)/java/java/function_cvxpy_max.java
package com.example;

import org.apache.commons.math3.stat.descriptive.rank.Max;
import java.util.Arrays;

public class CVXPYMaxFunction {
    public String compute_max_value(String vector) {
        try {
            
            String[] strArray = vector.replaceAll("[\\[\\]\\s]", "").split(",");

            
            double[] numbers = Arrays.stream(strArray)
                    .map(String::trim)
                    .mapToDouble(Double::parseDouble)
                    .toArray();

            
            if (Arrays.stream(numbers).anyMatch(Double::isNaN)) {
                throw new IllegalArgumentException("Input contains NaN values");
            }

            
            Max max = new Max();
            double maxValue = max.evaluate(numbers);

            
            return String.valueOf(maxValue);

        } catch (NumberFormatException e) {
            throw new IllegalArgumentException("Invalid input format.");
        }
    }
}

File: code/generate_strategies/Direct/python/function_cvxpy_max.py
import cvxpy as cp
import numpy as np

class CVXPYMaxFunction:
    def compute_max_value(self, vector: str) -> str:
        
        try:
            vector = [float(x) for x in vector.strip("[]").split(",")]
        except ValueError:
            raise ValueError("Invalid input format.")
        
        if any(np.isnan(v) for v in vector):
            raise ValueError("Input contains NaN values")
        
        max_value = cp.max(vector)

        max_value_value = max_value.value

        if max_value_value is None:
            raise ValueError("Failed to compute the max value.")
        
        return str(max_value_value)
